framework/include/utils/GenericRealU.h

```diff
@@ -0,0 +1,160 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Units.h"
+
+template <bool>
+class GenericRealU;
+
+// Port to std
+namespace std
+{
+// Power
+template <bool is_ad>
+GenericRealU<is_ad>
+pow(const GenericRealU<is_ad> & a, int b)
+{
+  return GenericRealU<is_ad>(pow(a.value(), b), pow(a.unit(), b));
+}
+
+// Absolute value
+template <bool is_ad>
+GenericRealU<is_ad>
+abs(const GenericRealU<is_ad> & a)
+{
+  return GenericRealU<is_ad>(abs(a.value()), a.unit());
+}
+}
+
+// The GenericReal with a const unit.
+template <bool is_ad>
+class GenericRealU
+{
+public:
+  /// Empty constructor
+  GenericRealU() : _value(0), _unit("1") {}
+
+  /// Construct from a value, set the unit to be dimensionless
+  GenericRealU(const GenericReal<is_ad> & value) : _value(value), _unit("1") {}
```

Comment:
rather than run the unit parser on "1", we shoufld have a pre-made constexpr unity unit.

Reply:
Yeah, originally I tried to construct a `MooseUnits` with a Real 1, hoping to get a `MooseUnits` with factor 1, shift 0 and empty base. But for some reason that didn't happen. I need to look into that.